src/app/components/weather-alert/weather-alert.component.ts

```diff
@@ -1,22 +1,35 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
 
 import {  WeatherService } from 'src/app/services/weather.service';
 import { WeatherAlertResponse } from 'src/app/models/weather-alert.model';
+import { interval, Subscription, take } from 'rxjs';
 
 @Component({
   selector: 'app-weather-alert',
   templateUrl: './weather-alert.component.html',
   styleUrls: ['./weather-alert.component.scss']
 })
-export class WeatherAlertComponent implements OnInit {
+export class WeatherAlertComponent implements OnInit, OnDestroy {
   @Input() weatherAlert: WeatherAlertResponse;
   public showToast: boolean = true; // By default we want to show the toast, but will be changed if the user closes the toast or if new weather alert data is called from the API
+  public toastTimeoutBarWidth: number = 100; // Default the bar to 100% width
+  private toastTimeoutTime: number = 10000; // Adjustable to make the toast display a shorter or longer amount of time
+  private toastTimeoutUpdateInterval: number = 10; // Adjustable to make the shrinking of the timer bar smoother in exchange for more processing power. The smaller the number, the more frequently the bar width will be calculated.
+  private toastLifespanEndTime: number;
+  private timeoutBarUpdateIntervalSubscription: Subscription;
```

Comment:
this variable and `toastTimeoutBarWidth` are the only one that are used outside of `setupIntervalTimer()`...no need to create a global variable for those

Reply:
I actually thought of that, but was on the fence about leaving it up here. The reason I decided is it's a sort of "settings" variable, so I left it up at the top so it could easily be located and tweaked. Probably wouldn't be too hard to find though, so I'll move it.

Reply:
toastTimeoutTime is used in setupIntervalTimer as well as timerToRemoveAlert so I'll make it a passable variable and declare in ngOnInit

Reply:
Moved `toastTimeoutTime` to ngOnInit, passed to both relevant methods, and moved other unecessarily global variables into `setupIntervalTimer()`. Also renamed `setupIntervalTimer` to have a more clear name.